samples/kotlin_compose/src/androidTest/java/io/customer/android/sample/kotlin_compose/MemoryLeakageInstrumentedTest.kt

```diff
@@ -0,0 +1,243 @@
+package io.customer.android.sample.kotlin_compose
+
+import android.app.Application
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performTextInput
+import androidx.test.core.app.ActivityScenario
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import io.customer.android.sample.kotlin_compose.data.repositories.PreferenceRepository
+import io.customer.android.sample.kotlin_compose.data.repositories.UserRepository
+import io.customer.android.sample.kotlin_compose.data.sdk.InAppMessageEventListener
+import io.customer.base.internal.InternalCustomerIOApi
+import io.customer.messaginginapp.MessagingInAppModuleConfig
+import io.customer.messaginginapp.ModuleMessagingInApp
+import io.customer.messagingpush.ModuleMessagingPushFCM
+import io.customer.sdk.CustomerIO
+import io.customer.sdk.util.CioLogLevel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import leakcanary.AppWatcher
+import leakcanary.DetectLeaksAfterTestSuccess
+import leakcanary.LeakAssertions
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
```

Comment:
Since this is an instrumentation test, at this time, we do not run this test automatically on the CI. 

We can and I believe we should. Mind creating a ticket to prioritize this?

Reply:
Good idea, here is the [PR](https://github.com/customerio/customerio-android/pull/233) to run instrumentation tests of sample apps as part of the workflow.